src/Analysis/pipeline/action_types.py

```diff
@@ -0,0 +1,453 @@
+        self.command = ""
+        """Command for popen"""
+        self.params = {}
+        """dictionary names => types of input ports"""
+        self.type =  type   
+
+__action_counter__ = 0
+"""action counter"""
+        
+class BaseActionType(metaclass=abc.ABCMeta):
+    """
+    Abstract class of action type, that define tasks method for
+    tasks classes
+    """
+
+    _name = ""
+    """Display name of action"""
+    _description = ""
+    """Display description of action"""
+
+    def __init__(self, **kwargs):
+        global __action_counter__
+ 
+        __action_counter__ += 1
+        self.id = __action_counter__
+        """unique action number"""
+        self.state = ActionStateType.created
+        """action state"""
+        self.inputs = []
+        """dictionary names => types of input ports"""
+        self.inputs_os = []
+        """
+        number of output slot where is action join (opposite slot)
+        default is first slot
+        """
+        """dictionary names => types of input ports"""
+        self.outputs = []
+        """dictionary names => DTT typse on output ports"""
+        self.variables = {}
+        """dictionary names => types of variables"""
+        self.type = ActionType.simple
+        """action type"""
+        for name, value in kwargs.items():
+            if name == 'Input':
+                self.inputs.append(value)
+            elif name == 'Inputs':
+                self.inputs.extend(value)
+            elif name == 'InputOpposite':
+                self.inputs_os.append(value)
+            elif name == 'InputOpposites':
+                self.inputs_os.extend(value)
+            elif name == 'Output':                
+                self.outputs.append(value)
+            elif name == 'Outputs':
+                self.outputs.extend(value)
+            else:
+                self.variables[name] = value
+        for i in range(len(self.inputs_os), len(self.inputs)):
+            self.inputs_os.append(0)
+
+    @abc.abstractmethod
+    def inicialize(self):
+        """inicialize action run variables"""
+        pass
+        
+    @abc.abstractmethod
+    def get_output(self, action, number):
+        """return output relevant for set action"""
+        pass
+   
+    def get_input_val(self, number):
+        """
+        if input is action type, return output from previous action,
+        else return input. Both action must be inicialized
+        """
+        if len(self.inputs)>number:
+            if isinstance(self.inputs[number],  BaseActionType):
+                return self.inputs[number].get_output(self, self.inputs_os[number])
+            else:
+                return self.inputs[number]                
+        return None
+    
+    @property
+    def name(self):
+        return self._name
+        
+    @property
+    def description(self):
+        return self._description
+    
+    @staticmethod
+    def _is_DTT(var):
+        """return if is var instance of DTT"""
```

Comment:
CompositeDTT by mělo být potomkem BaseDTT a pak by mělo stačit isinstance(var,BaseDTT) a na to nemusí být funkce,
pokud to nejde, měla by tato metoda být v BaseDTT.